infra/modules/aws-psoxy-lambda/main.tf

```diff
@@ -8,6 +8,16 @@ terraform {
   }
 }
 
+locals {
+  instance_ssm_prefix = coalesce(var.path_to_instance_ssm_parameters, "${upper(replace(var.function_name, "-", "_"))}_")
+
+  # parse PATH_TO_SHARED_CONFIG in super-hacky way
+  # expect something like:
+  # arn:aws:ssm:us-east-1:123123123123:parameter/PSOXY_SALT
+  salt_arn              = [for l in var.global_parameter_arns : l if endswith(l, "PSOXY_SALT")][0]
+  path_to_shared_config = regex("arn.+parameter/(.*)PSOXY_SALT", local.salt_arn)[0]
```

Comment:
so this is aggressively hacky ... could have added a variable to pass the `path_to_shared_config` in as a variable, but then what do if it's inconsistent with these arns? 

Could add validation check that `path_to_shared_config` is indeed in the ARNs, but that would ultimately look much like the regex I'm doing here. So imho this is the best thing to do until a `v0.5` where we avoid passing explicit ARNs, and require `path_to_shared_config` to be a path that each proxy instance should be able to read ALL descendants of.